src/views/creating_accommodation_facilities/AccommodationFacility.jsx

```diff
@@ -0,0 +1,319 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableContainer from '@material-ui/core/TableContainer';
+import TablePagination from '@material-ui/core/TablePagination';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+import Hidden from '@material-ui/core/Hidden';
+import Box from '@material-ui/core/Box';
+import { Card, CardContent, Typography } from '@material-ui/core';
+import {
+	getAccommodations,
+	getAccommodationsSuccess,
+} from '../../actions/accommodationFacilitiesActions';
+import { Styles } from '../../styles/userRole';
+import { createBrowserHistory } from 'history';
+import verifyToken from '../../helpers/tokenHelper';
+import { Toolbar } from '@material-ui/core';
+import Grid from '@material-ui/core/Grid';
+
+export const history = createBrowserHistory({
+	forceRefresh: true,
+});
+
+export function createData(name, description, location, category, owner, id) {
+	return { name, description, location, category, owner, id };
+}
+
+export class AccommodationFacility extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			rowsPerPage: 5,
+			page: 0,
+			tvalue: [],
+		};
+	}
+
+	UNSAFE_componentWillMount() {
+		const token = localStorage.getItem('token');
+		const user = verifyToken(token);
+		if (user.payload.role !== 'admin') {
+			history.push('/profile');
+		}
+		const { getAccommodations } = this.props;
+		getAccommodations();
+	}
```

Comment:
@mujohn26, I see the codes are okay. I just have a question, why are you redirecting the admin to the profile?

Reply:
@jo-rdan What I wanted for example if you try to copy the route and try to access it you  can't be able to access it unless you are admin it will direct you to the profile page instead of accessing it.

Reply:
Alright thanks for your clarifications